File: 프로그래머스/JavaScript/Lv. 1/42889. 실패율/실패율.js
function solution(N, stages) {
  let fail = [];
  let stageN = [];

  for (let i = 1; i < N + 1; i++) {
    let cnt = 0;
    for (let j = 0; j < stages.length; j++) {
      if (i == stages[j]) {
        cnt += 1;
      }
    }
    stageN.push(cnt);
  }

  let tmp = stages;
  for (let i = 0; i < N; i++) {
    fail.push(stageN[i] / tmp.length);
    tmp = tmp.filter((x) => x !== i + 1);
  }

  const answer = fail
    .map((value, index) => ({ value, index }))
    .sort((a, b) => b.value - a.value || a.index - b.index)
    .map((x) => x.index + 1);

  return answer;
}

File: 프로그래머스/Java/Lv. 1/42889. 실패율/실패율.java
class Solution { // Lv1. 실패율

	public int[] solution(int N, int[] stages) {

		int[] stage = new int[N + 2];

		for (int s : stages) {
			stage[s]++;
		}

		double[] fail = new double[N + 1];

		int member = stages.length;

		for (int i = 1; i <= N; i++) {
			if (member == 0) {
				fail[i] = 0;
			} else {
				fail[i] = (double) stage[i] / member;
			}

			member -= stage[i];
		}

		int[] answer = new int[N];

		for (int i = 0; i < N; i++) {

			double maxFail = -1;
			int max = 0;

			for (int j = 1; j <= N; j++) {
				if (fail[j] > maxFail) {
					maxFail = fail[j];
					max = j;
				}
			}

			answer[i] = max;
			fail[max] = -1;
		}

		return answer;
	}
}